pkg/api/testapi/testapi.go

```diff
@@ -24,63 +24,127 @@ import (
-// KUBE_API_VERSION env var.
-func Codec() runtime.Codec {
-	interfaces, err := latest.InterfacesFor(Version())
-	if err != nil {
-		panic(err)
+// Codec returns the codec for the API version of the group to test against, as
+// set by the KUBE_TEST_API env var.
+func Codec(group ...string) runtime.Codec {
+	if len(group) == 0 || group[0] == "" {
+		interfaces, err := latest.InterfacesFor(Version())
+		if err != nil {
+			panic(err)
+		}
+		return interfaces.Codec
 	}
-	return interfaces.Codec
+	if group[0] == "experimental" {
+		interfaces, err := explatest.InterfacesFor(Version(group...))
+		if err != nil {
+			panic(err)
+		}
+		return interfaces.Codec
+	}
+	panic(fmt.Errorf("cannot test group %s", group))
 }
 
-// Converter returns the api.Scheme for the API version to test against, as set by the
-// KUBE_API_VERSION env var.
-func Converter() runtime.ObjectConvertor {
-	interfaces, err := latest.InterfacesFor(Version())
-	if err != nil {
-		panic(err)
+// Converter returns the api.Scheme for the API version of the group to test
+// against, as set by the KUBE_TEST_API env var.
+func Converter(group ...string) runtime.ObjectConvertor {
+	if len(group) == 0 || group[0] == "" {
+		interfaces, err := latest.InterfacesFor(Version())
+		if err != nil {
+			panic(err)
+		}
+		return interfaces.ObjectConvertor
+	}
+	if group[0] == "experimental" {
+		interfaces, err := explatest.InterfacesFor(Version(group...))
+		if err != nil {
+			panic(err)
+		}
+		return interfaces.ObjectConvertor
 	}
-	return interfaces.ObjectConvertor
+	panic(fmt.Errorf("cannot test group %s", group))
 }
 
-// MetadataAccessor returns the MetadataAccessor for the API version to test against,
-// as set by the KUBE_API_VERSION env var.
-func MetadataAccessor() meta.MetadataAccessor {
-	interfaces, err := latest.InterfacesFor(Version())
-	if err != nil {
-		panic(err)
+// MetadataAccessor returns the MetadataAccessor for the API version of the
+// group to test against, as set by the KUBE_TEST_API env var.
+func MetadataAccessor(group ...string) meta.MetadataAccessor {
+	if len(group) == 0 || group[0] == "" {
+		interfaces, err := latest.InterfacesFor(Version())
+		if err != nil {
+			panic(err)
+		}
+		return interfaces.MetadataAccessor
+	}
+	if group[0] == "experimental" {
+		interfaces, err := explatest.InterfacesFor(Version(group...))
+		if err != nil {
+			panic(err)
+		}
+		return interfaces.MetadataAccessor
 	}
-	return interfaces.MetadataAccessor
+	panic(fmt.Errorf("cannot test group %s", group))
 }
 
-// SelfLink returns a self link that will appear to be for the version Version().
+// SelfLink returns a self link that will appear to be for the version
+// Version(group...).
 // 'resource' should be the resource path, e.g. "pods" for the Pod type. 'name' should be
 // empty for lists.
-func SelfLink(resource, name string) string {
+func SelfLink(resource, name string, group ...string) string {
```

Comment:
I feel like group should actually be the first parameter in this function. The more I look at this file, the less I like making group an optional parameter.